modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v5_4_4/DDMStructureUpgradeProcess.java

```diff
@@ -51,18 +51,21 @@ private void _upgradeDDMStructure() throws Exception {
 		try (PreparedStatement selectPreparedStatement =
 				connection.prepareStatement(
 					StringBundler.concat(
-						"select DDMStructure.structureId, ",
+						"select DDMStructure.structureId,",
+						"DDMStructure.ctCollectionId, ",
 						"DDMStructureVersion.definition from DDMStructure ",
 						"inner join DDMStructureVersion on ",
 						"DDMStructure.structureid = ",
-						"DDMStructureVersion.structureid where ",
+						"DDMStructureVersion.structureid and ",
+						"DDMStructure.ctCollectionId = ",
+						"DDMStructureVersion.ctCollectionId where ",
```

Comment:
Can you test if it is necessary to do this here?

Reply:
Hi @marcelabc !
Upon the removal of this code the `DDMStructureCTUpgradeProcessTest` failed. I think this code is truly necessary then.

Reply:
Since you verify, can you sort it please?